containers/test-apps/kitchen/bin/kitchen

```diff
@@ -599,6 +612,24 @@ class Kitchen(HTTPWebSocketsHandler):
     def __process_headers(self):
         """Handle logic for all supported Kitchen HTTP header values."""
         assert self.__path is not None, 'Processes headers AFTER processing the request path.'
+        global _default_response_status
+
+        # Temporarily set status on responses
+        default_status_value = self.headers.get('x-kitchen-default-status-value')
+        if default_status_value is not None:
+            if _allow_response_status_change:
+                self.logger().info('Changing default response status to {}'.format(default_status_value))
+                _default_response_status = int(default_status_value)
+                self.__status = _default_response_status
```

Comment:
Since `__process_headers` comes after `__process_path`, this line would clobber the `/bad-status` endpoint's status. If the default status logic is moved to `send_response` as described above, I think the interaction with `/bad-status` would be more intuitive.

Reply:
Moved default status processing and setting of `__status` to the end of `__process_headers`